utils/include/Utils.cpp

```diff
@@ -238,3 +239,72 @@ EMemoryConvertType Utils::DefineCountType(int count_type_from_xml)
         return EMemoryConvertType::BYTES;
     }
 }
+
+bool Utils::StringToDate(const std::string& date_str,
+                         const std::string& date_format, time_t& result)
+{
+    tm temp_date;
+    std::istringstream ss_date(date_str);
+    ss_date >> std::get_time(&temp_date, date_format.c_str());
+    if(ss_date.fail())
+    { return false;}
+
+    result = mktime(&temp_date);
+    if(result < 0)
+    { return false;}
+
+    return true;
+}
+
+bool Utils::TryCreateDirectory(const std::string& path,
+    std::filesystem::perms permission,
+    std::filesystem::perm_options perms_action)
+{
+    if (std::filesystem::create_directory(path))
+    {
+        //for windows will be a little different
+        std::filesystem::permissions(path, permission, perms_action);
+        return true;
+    }
+
+    return std::filesystem::is_directory(path);
+}
+
+bool Utils::IsDayPassed(std::string& day)
+{
+    std::string curr_day;
+    if (Utils::TryGetCurrentDateAndTimeFormatted(curr_day, "%d"))
```

Comment:
Is Utils::TryGetCurrentDateAndTimeFormatted the same as TryGetCurrentDateAndTimeFormatted?

Reply:
Utils:: deleted